usr/share/rear/skel/default/etc/scripts/system-setup.d/55-migrate-network-devices.sh

```diff
@@ -30,14 +30,38 @@ ORIGINAL_DEVICES=()
 while read orig_dev orig_mac junk ; do
     ORIGINAL_DEVICES=( "${ORIGINAL_DEVICES[@]}" "$orig_dev")
     ORIGINAL_MACS=( "${ORIGINAL_MACS[@]}" "$orig_mac" )
-    # Continue with the next original MAC address if it is found on the current system:
-    ip link show | grep -q "$orig_mac" && continue
-    MIGRATE_DEVNAMES=( "${MIGRATE_DEVNAMES[@]}" "$orig_dev" )
-    MIGRATE_MACS=( "${MIGRATE_MACS[@]}" "$orig_mac" )
+    # Continue with the next original MAC address if it is found on the current
+    # system, otherwise we consider it needs migration:
+    new_dev=$( get_device_by_hwaddr "$orig_mac" )
+    if [ -n "$new_dev" ] ; then
+        [ "$new_dev" = "$orig_dev" ] && continue
+        # The device was found but has been renamed (it was customized in
+        # source system).
+        # Create a temporary mac mapping, we don't want finalize() to update
+        # the ifcfg-* files!
+        echo "$orig_mac $orig_mac $orig_dev" >> /tmp/mac
```

Comment:
Please use a more explanatory file name than `/tmp/mac`
(the `/etc/rear/mappings/mac` is not really good but o.k. because its path tells what it is)
preferably via a meaningful variable name like
```
local tmp_mac_mapping_file="/tmp/rear_mappings_mac"
```
in relation to `MAC_MAPPING_FILE=/etc/rear/mappings/mac`

I think in 'skel' scripts (that run only during recovery system startup phase)
it is o.k to use a hardcoded `/tmp/` instead of $TMPDIR
nevertheless I wonder if we might better also use $TMPDIR here?
Or is using $TMPDIR here perhaps useless because during recovery system
startup phase '/tmp/' is the only available place where to store temporary files
or is using $TMPDIR here perhaps even impossible because $TMPDIR
is not set during recovery system startup phase?

Reply:
@rmetrich rear code should never write directly under `/tmp` itself as it has its own TMP_DIR, please keep this in mind. There is already enough garbage under /tmp that no-one cleans up - we should not promote this either ;-)
If it is about a temp map file why not use temp in the name to clearly identify it is not a final mapping file?

Reply:
@gdha 
I think ReaR has its own TMP_DIR only when "sbin/rear" is called
(I find `TMP_DIR=` only in usr/sbin/rear as `TMP_DIR=$BUILD_DIR/tmp`)
but this pull request is about
skel/default/etc/scripts/system-setup.d/55-migrate-network-devices.sh
that is (as far as I know) only run during recovery system startup phase
i.e. it is not run/sourced from within a running "sbin/rear", cf. my above
https://github.com/rear/rear/pull/2004#discussion_r241683384